File: sdks/java/src/main/java/bridgebank/api/ApiRequestBuilder.java
package bridgebank.api;

import bridgebank.api.bancos.BancosRequestBuilder;
import bridgebank.api.classificacao.ClassificacaoRequestBuilder;
import bridgebank.api.enums.EnumsRequestBuilder;
import bridgebank.api.extratos.ExtratosRequestBuilder;
import bridgebank.api.pagamentos.PagamentosRequestBuilder;
import bridgebank.api.reconciliacao.ReconciliacaoRequestBuilder;
import com.microsoft.kiota.BaseRequestBuilder;
import com.microsoft.kiota.RequestAdapter;
import java.util.HashMap;
import java.util.Objects;
/**
 * Builds and executes requests for operations under /api
 */
@jakarta.annotation.Generated("com.microsoft.kiota")
public class ApiRequestBuilder extends BaseRequestBuilder {
    /**
     * The bancos property
     * @return a {@link BancosRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public BancosRequestBuilder bancos() {
        return new BancosRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The classificacao property
     * @return a {@link ClassificacaoRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ClassificacaoRequestBuilder classificacao() {
        return new ClassificacaoRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The enums property
     * @return a {@link EnumsRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public EnumsRequestBuilder enums() {
        return new EnumsRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The extratos property
     * @return a {@link ExtratosRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ExtratosRequestBuilder extratos() {
        return new ExtratosRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The pagamentos property
     * @return a {@link PagamentosRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public PagamentosRequestBuilder pagamentos() {
        return new PagamentosRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * The reconciliacao property
     * @return a {@link ReconciliacaoRequestBuilder}
     */
    @jakarta.annotation.Nonnull
    public ReconciliacaoRequestBuilder reconciliacao() {
        return new ReconciliacaoRequestBuilder(pathParameters, requestAdapter);
    }
    /**
     * Instantiates a new {@link ApiRequestBuilder} and sets the default values.
     * @param pathParameters Path parameters for the request
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ApiRequestBuilder(@jakarta.annotation.Nonnull final HashMap<String, Object> pathParameters, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/api", pathParameters);
    }
    /**
     * Instantiates a new {@link ApiRequestBuilder} and sets the default values.
     * @param rawUrl The raw URL to use for the request builder.
     * @param requestAdapter The request adapter to use to execute the requests.
     */
    public ApiRequestBuilder(@jakarta.annotation.Nonnull final String rawUrl, @jakarta.annotation.Nonnull final RequestAdapter requestAdapter) {
        super(requestAdapter, "{+baseurl}/api", rawUrl);
    }
}

File: sdks/python/bridgebank/api/api_request_builder.py
from __future__ import annotations
from collections.abc import Callable
from kiota_abstractions.base_request_builder import BaseRequestBuilder
from kiota_abstractions.get_path_parameters import get_path_parameters
from kiota_abstractions.request_adapter import RequestAdapter
from typing import Any, Optional, TYPE_CHECKING, Union

if TYPE_CHECKING:
    from .bancos.bancos_request_builder import BancosRequestBuilder
    from .classificacao.classificacao_request_builder import ClassificacaoRequestBuilder
    from .enums.enums_request_builder import EnumsRequestBuilder
    from .extratos.extratos_request_builder import ExtratosRequestBuilder
    from .pagamentos.pagamentos_request_builder import PagamentosRequestBuilder
    from .reconciliacao.reconciliacao_request_builder import ReconciliacaoRequestBuilder

class ApiRequestBuilder(BaseRequestBuilder):
    """
    Builds and executes requests for operations under /api
    """
    def __init__(self,request_adapter: RequestAdapter, path_parameters: Union[str, dict[str, Any]]) -> None:
        """
        Instantiates a new ApiRequestBuilder and sets the default values.
        param path_parameters: The raw url or the url-template parameters for the request.
        param request_adapter: The request adapter to use to execute the requests.
        Returns: None
        """
        super().__init__(request_adapter, "{+baseurl}/api", path_parameters)
    
    @property
    def bancos(self) -> BancosRequestBuilder:
        """
        The bancos property
        """
        from .bancos.bancos_request_builder import BancosRequestBuilder

        return BancosRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def classificacao(self) -> ClassificacaoRequestBuilder:
        """
        The classificacao property
        """
        from .classificacao.classificacao_request_builder import ClassificacaoRequestBuilder

        return ClassificacaoRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def enums(self) -> EnumsRequestBuilder:
        """
        The enums property
        """
        from .enums.enums_request_builder import EnumsRequestBuilder

        return EnumsRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def extratos(self) -> ExtratosRequestBuilder:
        """
        The extratos property
        """
        from .extratos.extratos_request_builder import ExtratosRequestBuilder

        return ExtratosRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def pagamentos(self) -> PagamentosRequestBuilder:
        """
        The pagamentos property
        """
        from .pagamentos.pagamentos_request_builder import PagamentosRequestBuilder

        return PagamentosRequestBuilder(self.request_adapter, self.path_parameters)
    
    @property
    def reconciliacao(self) -> ReconciliacaoRequestBuilder:
        """
        The reconciliacao property
        """
        from .reconciliacao.reconciliacao_request_builder import ReconciliacaoRequestBuilder

        return ReconciliacaoRequestBuilder(self.request_adapter, self.path_parameters)
    

